src/ophyd_async/core/_utils.py

```diff
@@ -342,7 +347,7 @@ def __call__(self) -> T:
         return self._obj
 
 
-class LazyMock:
+class DeviceMock(Generic[DeviceType]):
```

Comment:
Move `DeviceMock` and all other additions to this file into `_device.py` instead